### PR TITLE
libft: mapss operations, ft_free_s, ft_if

### PR DESCRIPTION
**Pull request by ioannnikulin**

### CODE REVIEW

File: libft/sources/map/mapss/ft_mapss_get.c [hunk 1]
```diff
@@ -19,9 +19,9 @@ char	*ft_mapss_get(t_mapss *map, char *key)
 	if (!map || !map->head || !key)
 		return (0);
 	cur = map->head;
-	while (ft_strcmp(key, ((t_mapss_entry*)cur)->key) < 0)
+	while (cur && ft_strcmp(key, ((t_mapss_entry*)cur->content)->key) < 0)
```
Comment (cr3mebrulee) [inline libft/sources/map/mapss/ft_mapss_get.c:22]: maybe we need to change here for > 0, i'm not sure, let me run one more test.
are you sure in this condition?

Comment (ioannnikulin) [reply libft/sources/map/mapss/ft_mapss_get.c:22]: `if (cur)` is basically the same as `if (cur > 0)` considering how 0/1/false/true are treated in C. Or you mean the strcmp result? Then definitely not, we scroll down the list WHILE what we have in hands is LESS than what we see in the catalogue. Once we hit >=, this means that either we've found what we were looking for, or we missed its place, so it was not in the list at all.

File: libft/sources/map/mapss/ft_mapss_del.c [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	free(((t_mapss_entry *)node->content)->value);
+	free((t_mapss_entry *)node->content);
+	free(node);
+}
+
+int	ft_mapss_del(t_mapss *map, const char *key)
+{
+	t_dlist	*cur_node;
+	char	*cur_key;
+
```
[the remark this answers is not in the document]
Comment (ioannnikulin) [reply libft/sources/map/mapss/ft_mapss_del.c:29]: Can do. Updated, thank you. Don't want to check map->head, having an empty map is a pretty standard situation, it goes into the pile of "not found, return 1"

File: libft/sources/map/mapss/ft_mapss_get.c (hunk 1 above)
Comment (ioannnikulin) [reply libft/sources/map/mapss/ft_mapss_get.c:22]: Also if you're making tests, consider adding them to libft test suite, or at least listing here or in a separate issue. We are starting to lack tests sorely.

Comment (ioannnikulin) [reply libft/sources/map/mapss/ft_mapss_get.c:22]: Oops, right

Comment (ioannnikulin) [reply libft/sources/map/mapss/ft_mapss_get.c]: fixed

**Review by cr3mebrulee — APPROVED** [review]
thank you

**Pull request merged**